htmd/parameterization/ffmolecule.py

```diff
@@ -219,137 +193,71 @@ def _fitDihedral_objective(self, x):
+        point_file = os.path.join(espDir, "00000", "grid.dat")
+        if os.path.exists(point_file):
+            # Load a point file if one exists from a previous job
+            esp_points = np.loadtxt(point_file)
+            print("Reusing previously-generated point cloud")
+        else:
+            # Generate ESP points
+            esp_points = ESP.generate_points(self)[0]
 
+        # Run QM simulation
         self.qm.molecule = self
-        self.qm.esp_points = points
+        self.qm.esp_points = esp_points
         self.qm.optimize = False
         self.qm.restrained_dihedrals = None
-        self.qm.directory = espdir
-        results = self.qm.run()
-        if results[0].errored:
+        self.qm.directory = espDir
+        qm_results = self.qm.run()
+        if qm_results[0].errored:
             raise RuntimeError("QM Calculation failed")
 
-        esp_grid = results[0].esp_points
-        esp = results[0].esp_values
-        self.coords = results[0].coords
+        # Safeguard QM code from changing coordinates :)
+        assert np.all(np.isclose(self.coords, qm_results[0].coords, atol=1e-6))
 
-        #    print(results[0].dipole )
-        #    print(results[0].quadrupole )
-        #    print(results[0].mulliken )
+        # Fit ESP charges
+        self.esp = ESP()
+        self.esp.molecule = self
+        self.esp.qm_results = qm_results
+        self.esp.fixed = fixed
+        esp_result = self.esp.run()
+        esp_charges, esp_loss = esp_result['charges'], esp_result['loss']
 
-        self._fitCharges_grid = esp_grid
-        self._fitCharges_esp = esp
+        # Update the charges
+        self.charge[:] = esp_charges
+        self._rtf.updateCharges(esp_charges)
 
-        # set up the restraints to fit
+        return esp_loss, qm_results[0].dipole
 
-        N = len(self._equivalent_atom_groups)  # - 1
-        lb = np.ones((N)) * -1.25
-        ub = np.ones((N)) * +1.25
+    def getDipole(self):
+        """Calculate the dipole moment (in Debyes) of the molecule"""
 
-        # Fix the charges of the specified atoms to those already set in the 
-        # charge array. Note this also fixes the charges of the atoms in the
-        # same equivalency group.
-        # 
-        for atom in fixed:
-            group = self._equivalent_group_by_atom[ atom ]
-            lb[group] = self.charge[atom] 
-            ub[group] = self.charge[atom] 
+        coords = self.coords[:, :, self.frame] - self.centreOfMass
 
-        # If the restraint relates to an H, set the lower bound to 0
-        for i in range(N):
-            if "H" == self.element[self._equivalent_atom_groups[i][0]]:
-                lb[i] = 0.001
+        dipole = np.zeros(4)
+        dipole[:3] = np.dot(self.charge, coords)
+        dipole[3] = np.linalg.norm(dipole[:3]) # Total dipole moment
+        dipole /= 0.20819434 # e * Ang --> Debye (https://en.wikipedia.org/wiki/Debye)
```

Comment:
There aren't Debyes in pint, but there is statCoulomb. Tomorrow I do this change. I'll also propose a PR to pint.